### DISCUSSION TITLE
Anaconda crashes when using ‘autostep’ in Kickstart file

### DISCUSSION THREAD

**Discussion by century6** [rhinstaller/anaconda#3347] [Q&A] [answered]
After I used the autostep functionality on anaconda-33.16.3.26, an error occurred during the installation process, and the phenomenon of the problem is consistent with [rhbz#1882538](https://bugzilla.redhat.com/show_bug.cgi?id=1882538). Does the community plan to repair the problem?

In addition, I found that the autostep functionality was removed in the [MR,](https://github.com/rhinstaller/anaconda/pull/2978) but the MR was only synchronized to anaconda-34.12-1 or above. Why does the autostep function still exist in versions below anaconda-34.12-1?

**Comment by VladimirSlavik** [comment, marked as answer]
Hi century6!

> After I used the autostep functionality on anaconda-33.16.3.26, an error occurred during the installation process, and the phenomenon of the problem is consistent with [rhbz#1882538](https://bugzilla.redhat.com/show_bug.cgi?id=1882538). Does the community plan to repair the problem?

I can't speak for the community, but the core developer team will not, because it is "not possible". Read below for details.

> In addition, I found that the autostep functionality was removed in the [MR,](https://github.com/rhinstaller/anaconda/pull/2978) but the MR was only synchronized to anaconda-34.12-1 or above. Why does the autostep function still exist in versions below anaconda-34.12-1?

- Anaconda versions correspond to Fedora versions. Eg. 33.12 is for Fedora 33.

- Bugs reported before Fedora N goes GA are in Anaconda version N.M, so they can be fixed in Anaconda N.M+1 and all is well. Hoewever this is a pre-release of that Fedora version, so "normal users" don't usually try this.

- Once any given Fedora version goes GA, the installation images are not updated again. Updated content will reach users either by running `dnf update` or  by downloading packages from the updates repository during installation. **Updates to Anaconda N can not reach users after Fedora N is released as GA**. Thus, there is no point in maintaining versions where the corresponding Fedora got released as GA.

- Most users do not need anaconda in their daily life. They encounter only the particular version of Anaconda that is "cemented" in the GA images.

- After a Fedora version N (eg. 33) goes GA, users report tons of bugs against anaconda there. These can be fixed earliest in anaconda N+1 (34 respectively).

- Tools that handle image creation - lorax, livemedia creator, weldr, etc. - use Anaconda too. They get it from a package, so the "no updates" problem does not apply to them. If this machinery needs some fixes, we do make and release them, but that's pretty rare.

Hopefully that explains why the answer is "no" here.

Do feel free to ask for more details, clarifications etc. Or maybe describe why you need this fix on 33? Perhaps there's a way to work around that.

----

The version and bug you referenced are from RHEL. If you want this fixed in RHEL, the above needs a slight correction. Anaconda in RHEL uses a different version number scheme. 33.16.3.26 means it is based on the upstream (Fedora) version 33.16, and the next number 3 is for RHEL 8.3 (4 = 8.4 etc.) With that in mind, the rest should be mostly true - I believe we don't update RHEL media either (not 100% certain though).
